documentation/docs/20-core-concepts/20-load.md

```diff
@@ -420,19 +420,23 @@ export function load({ locals }) {
 
 In the browser, you can also navigate programmatically outside of a `load` function using [`goto`](modules#$app-navigation-goto) from [`$app.navigation`](modules#$app-navigation).
 
-## Promise unwrapping
+## Promises
 
-Top-level promises will be awaited, which makes it easy to return multiple promises without creating a waterfall:
+Promises at the _top level_ of the returned object will be awaited, making it easy to return multiple promises without creating a waterfall. When using a server `load`, _nested_ promises will be streamed to the browser as they resolve:
```

Comment:
I think some additional context as to _why_ you would want to do this would be helpful. Also, do we have guidance for when not to do this? (since it affects the SSR'd page)

```suggestion
Promises at the _top level_ of the returned object will be awaited, making it easy to return multiple promises without creating a waterfall. When using a server `load`, _nested_ promises will be streamed to the browser as they resolve. This is useful if you have slow, non-essential data, since you can start rendering the page before the `fetch` call resolves:
```

Reply:
Copying over some potential guidance from another conversation:

> I wonder if we want to draw a 'essential/non-essential' distinction rather than 'fast/slow' (not least because streaming requires JS)? E.g. your blog post is essential — if [slug] doesn't exist in the database, that's a 404, if it fails to load it's a 500 — but the comments might be non-essential

Reply:
I like that, though the bit about `fetch` implies that it only applies to client-side nav rather than the initial page render. Maybe this?

```suggestion
Promises at the _top level_ of the returned object will be awaited, making it easy to return multiple promises without creating a waterfall. When using a server `load`, _nested_ promises will be streamed to the browser as they resolve. This is useful if you have slow, non-essential data, since you can start rendering the page before all the data is available:
```